src/torchcodec/decoders/_core/VideoDecoderOps.cpp

```diff
@@ -168,7 +168,7 @@ void _add_video_stream(
   if (device.has_value()) {
     if (device.value() == "cpu") {
       options.device = torch::Device(torch::kCPU);
-    } else if (device.value().starts_with("cuda")) {
+    } else if (device.value().rfind("cuda", 0) == 0) { // starts with "cuda"
```

Comment:
Since `rfind()` starts from the end ("reverse find"), this is technically "ends with 'cuda'". See https://en.cppreference.com/w/cpp/string/basic_string/rfind.

Reply:
rfind(*, 0) looks for a prefix